terra/task.py

```diff
@@ -76,6 +78,9 @@ def __call__(self, *args, **kwargs):
         terra.logger._logs.reconfigure_logger()
         return_value = self.run(*args, **kwargs)
     else:
+      original_zone = settings.terra.zone
```

Comment:
Would be nice to do using `with settings`. Perhaps  if Settings/ObjectDict ever supports COW

Reply:
Also, can self.run throw an exception? If so, `settings.terra.zone` won't get reset